packages/ui/src/components/form-primitives/select.tsx

```diff
@@ -87,6 +87,8 @@ interface SelectProps<T = string>
   contentClassName?: string
   suffix?: ReactNode
   triggerClassName?: string
+  triggerRenderer?: (selectedLabel?: ReactNode) => ReactNode
```

Comment:
Select component should not have a custom renderer for the trigger IMO. We can use the Dropdown for that.
Instead, maybe we should do a `variant` prop on Select and support a limited set of trigger designs powered via tokens or existing Buttons.
cc @praneshg239 @3em what do you think?

Reply:
IMO, It is still achievable through `Dropdown` component, also `Dropdown` component natively comes with `checkmark` prop that adds check icon to the option. 

We don't want to alter `Select` component with trigger. It is like reinventing the wheel. 

@abhinavrastogi-harness thoughts?